mathics/builtin/base.py

```diff
@@ -611,6 +611,8 @@ def get_string_value(self):
 
     def sameQ(self, expr) -> bool:
         """Mathics SameQ"""
+        if isinstance(expr, BoxConstruct):
```

Comment:
When a general method like `SameQ` is asking about a specific kind of thing like `BoxConstruct` that should be a harbinger that there is a design problem and that a general concept is missing. 

At a minimum, a lot of explaining is needed as to why sameQ which is testing whether something is identical needs to understand why something is a Boxing construct. 

Sometimes instead there is a fundamental concept that needs to be made explicit.

Reply:
This is because if `BoxConstruct` compares against other kind of expression, it can delegate the implementation to the other object. But if `expr` is also a `BoxConstruct`, this would produce an infinite loop. However, in this case, we just need to compare if `self` and `expr` are the same Python object.